### PR TITLE
MO dev guide refactoring (#3266)

### PR DESCRIPTION
**Pull request by lazarevevgeny**
Cherry-pick of changes from https://github.com/openvinotoolkit/openvino/pull/3266 + several refactoring changes to re-use template extension for the "Custom Operations Guide"

* Updated MO extension guide

* Minor change and adding svg images

* Added additional information about operation extractors. Fixed links and markdown issues

* Added missing file with information about Caffe Python layers and image for MO transformations dependencies graph

* Added section with common graph transformations attributes and diagram with anchor transformations. Added list of available front phase transformations

* Added description of front-phase transformations except the scope-defined and points defined. Removed legacy document and examples for such transformations.

* Added sections about node name pattern defined front phase transformations. Copy-pasted the old one for the points defined front transformation

* Added description of the rest of front transformations and and all middle and back phase transformations

* Refactored Legacy_Mode_for_Caffe_Custom_Layers and updated the Customize_Model_Optimizer with information about extractors order

* Added TOC for the MO Dev guide document and updated SVG images with PNG ones

* Fixed broken link. Removed redundant image

* Fixed broken links

* Added information about attributes 'run_not_recursively', 'force_clean_up' and 'force_shape_inference' of the transformation

* Code review comments

* Added a section about `Port`s

* Extended Ports description with examples

* Added information about Connections

* Updated MO README.md and removed a lot of redundant and misleading information

* Updates to the Customize_Model_Optimizer.md

* More updates to the Customize_Model_Optimizer.md

* Final updates for the Customize_Model_Optimizer.md

* Fixed some broken links

* More fixed links

* Refactored Custom Layers Guide: removed legacy and incorrect text, added up-to-date.

* Draft implementation of the Custom layer guide example for the MO part

* Fixed broken links using #. Change layer->operation in extensibility documents

* Updated Custom operation guide with IE part

* Fixed broken links and minor updates to the Custom Operations Guide

* Updating links

* Layer->Operation

### CODE REVIEW

File: docs/HOWTO/ie_cpu_extension/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# Contains implementation of the basic layer methods
+#
+# Refer to the section "Adding Your Own Kernels to the Inference Engine" in
+# OpenVINO* documentation (either online or offline in
+# <INSTALL_DIR>/deployment_tools/documentation/docs/index.html an then navigate
+# to the corresponding section).
+# ===============================================================================
+
+# [fft_cmake_list:cmake]
+set(CPU_EXTENSIONS_BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR} CACHE INTERNAL "")
```
Comment (ilya-lavrenov) [inline docs/HOWTO/ie_cpu_extension/CMakeLists.txt:17]: we already have `template_extension` inside `docs` folder. Do we need one more?

Comment (ilya-lavrenov) [reply docs/HOWTO/ie_cpu_extension/CMakeLists.txt:17]: BTW, if you refer to this extensions in your docs, it seems you have a duplication with https://docs.openvinotoolkit.org/latest/openvino_docs_IE_DG_Extensibility_DG_Intro.html

CC @ilyachur

File: docs/HOWTO/Custom_Layers_Guide.md [hunk 2]
```diff
@@ -1,200 +1,362 @@
-# Custom Layers Guide {#openvino_docs_HOWTO_Custom_Layers_Guide}
+# Custom Operations Guide {#openvino_docs_HOWTO_Custom_Layers_Guide}
+
```
Comment (ilya-lavrenov) [inline docs/HOWTO/Custom_Layers_Guide.md:2]: Looks like this document and https://github.com/openvinotoolkit/openvino/blob/master/docs/IE_DG/Extensibility_DG/Intro.md are about the same. Do we need this one? 

I suppose we need to **merge** them at least.

File: docs/HOWTO/ie_cpu_extension/CMakeLists.txt (hunk 1 above)
Comment (lazarevevgeny) [reply docs/HOWTO/ie_cpu_extension/CMakeLists.txt:17]: This file is a part of the sample for the "Custom operation guide" which allows us to run one specific model. And I use different "tags" so there is no conflict/duplication.

File: docs/HOWTO/Custom_Layers_Guide.md (hunk 2 above)
Comment (lazarevevgeny) [reply docs/HOWTO/Custom_Layers_Guide.md:2]: This guide covers the whole pipeline of the operation enabling including MO and IE. The https://github.com/openvinotoolkit/openvino/blob/master/docs/IE_DG/Extensibility_DG/Intro.md is about extending IE part only. There was a specific request from marketing to have a guide which covers the whole story of operation enabling in one place, so I cannot merge the guides.

File: docs/HOWTO/ie_cpu_extension/CMakeLists.txt (hunk 1 above)
Comment (ilyachur) [reply docs/HOWTO/ie_cpu_extension/CMakeLists.txt:17]: I agree with Ilya, why cannot we extend `template_extension` library for these cases?
And add new tests to test these kernels.

We have tests for `template_extension` library but looks like we don't validate this code.

Comment (lazarevevgeny) [reply docs/HOWTO/ie_cpu_extension/CMakeLists.txt]: Done

**Review by ilya-lavrenov — APPROVED** [review]

File: docs/IE_DG/Extensibility_DG/Extension.md [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # Extension Library {#openvino_docs_IE_DG_Extensibility_DG_Extension}
 
 Inference Engine provides an InferenceEngine::IExtension interface, which defines the interface for Inference Engine Extension libraries.
-All extension libraries should be inherited from this interface.
+All extension libraries should be inherited from this interface. The example below contains implementation of two operations: `Template`
+used as an example in this document and `FFT` used as a more complex example from the [Custom Operations Guide](../../HOWTO/Custom_Layers_Guide.md).
```
Comment (ilyachur) [inline docs/IE_DG/Extensibility_DG/Extension.md:5]: Can we highlight here that FFT kernel requires OpenCV?

**Review by ilyachur — COMMENTED** [review]
Can you add new test to check that FFT kernel works as expected?

File: docs/IE_DG/Extensibility_DG/Extension.md (hunk 3 above)
Comment (lazarevevgeny) [reply docs/IE_DG/Extensibility_DG/Extension.md:5]: Done

**Pull request merged**